scalingo/data_scalingo_addon_provider.go

```diff
@@ -0,0 +1,165 @@
+package scalingo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/Scalingo/go-scalingo/v5"
+)
+
+func dataSourceScAddonProvider() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceScAddonProviderRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"short_description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"category": {
+				Type:     schema.TypeMap,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"provider_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"provider_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"plans": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"display_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"price": {
+							Type:     schema.TypeFloat,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"position": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"on_demand": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"disabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"disabled_alternative_plan_id": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"sku": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"hds_available": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceScAddonProviderRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	name, _ := d.Get("name").(string)
+
+	addonProviders, err := client.AddonProvidersList(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var selected *scalingo.AddonProvider
+
+	for _, v := range addonProviders {
+		if v.Name == name {
+			selected = v
+			break
+		}
+	}
+
+	if selected == nil {
+		return diag.Errorf("addon provider '%v' not found", name)
+	}
+	d.SetId(selected.ID)
+	categoryMap := map[string]interface{}{
```

Comment:
suggestion: for the same reason as `planMap`

```suggestion
	category := map[string]interface{}{
```

Reply:
You don't want to take this one into account? :'(